src/pyuvsim/simsetup.py

```diff
@@ -2078,6 +2075,7 @@ def initialize_uvdata_from_params(
         do_blt_outer=True,
         time_axis_faster_than_bls=False,
         antpairs=antpairs,
+        check_kw=check_kw,
```

Comment:
In #503 `initialize_uvdata_from_keywords` was updated to pass `check_kw` to this function (`initialize_uvdata_from_params`) with the justification that the UVData.new method can do the checks, so they don't need to be done in `_complete_uvdata`. However, we'd already marked `check_kw` in `initialize_uvdata_from_params` as deprecated and having no effect. So I revived it here and passed it into UVData.new(). However I could also support just removing it so that all checks are run when we make a new object.

Reply:
I definitely do not want to have to do all checks when making a new object, because in practice we often split simulations into multiple parts, each with its own config YAML, and its overkill to check object creation for each, when each is just changing e.g. the frequency channel simulated. When each individual simulation is somewhat small (one channel) the setup time can be a non-negligible overhead.

Reply:
So, keeping this argument somewhere is a good idea.